src/api/procedures/createSecurityToken.ts

```diff
@@ -236,14 +252,29 @@ export function getAuthorization(
     transactions.push(TxTags.asset.RegisterCustomAssetType);
   }
 
-  return {
-    roles: [{ type: RoleType.TickerOwner, ticker }],
+  let roles: Role[] = [];
+
+  const reservation = new TickerReservation({ ticker }, context);
+
+  const { status } = await reservation.details();
```

Comment:
This is also being fetched in the prepare method. There is a way to only need to fetch it once. Check out `prepareStorage` in `configureDividendDistribution` to see how it can be done, but the basic idea is that procedures can have a storage where common values can be pre-fetched to be used in both the prepare and getAuthorization methods